Libraries/SDHC/Source/sdhc_lib.c

```diff
@@ -159,11 +159,20 @@ int MXC_SDHC_Lib_GetCSD(mxc_sdhc_csd_regs_t *csd)
 }
 
 /* ************************************************************************** */
-unsigned int MXC_SDHC_Lib_GetCapacity(mxc_sdhc_csd_regs_t* csd)
+unsigned long long MXC_SDHC_Lib_GetCapacity(mxc_sdhc_csd_regs_t* csd)
 {
     unsigned int size = csd->csd.c_size;
 
-    return (size*(512*1024));
+    return ((unsigned long long)size)*((unsigned long long)512*1024);
+}
+
+/* ************************************************************************** */
+unsigned int MXC_SDHC_Lib_GetSectors(mxc_sdhc_csd_regs_t* csd, int i)
+{
+    if(i == 0) {
+        return csd->csd.sector_size_0;
```

Comment:
From the SD card spec, the "sector_size" field is fixed to 0x7F for SDHC and this field should not be used.

In this function we simply want to retrieve the "c_size" field to get the number of sectors on the card.  The spec indicates that it is 22 bits wide, so unsigned int should be fine as the return type.

I suggest

```C
unsigned int MXC_SDHC_Lib_GetSectors(mxc_sdhc_csd_regs_t* csd) {
    return csd->csd.c_size;
}
```

Reply:
changed